File: src/main/java/um/haberes/report/model/haberes/core/CargoLiquidacionDto.java
package um.haberes.report.model.haberes.core;

import com.fasterxml.jackson.annotation.JsonFormat;
import java.math.BigDecimal;
import java.time.OffsetDateTime;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CargoLiquidacionDto {

    private Long cargoLiquidacionId;

    private Long legajoId;

    private Integer anho;

    private Integer mes;

    private Integer dependenciaId;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssXX", timezone = "UTC")
    private OffsetDateTime fechaDesde;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssXX", timezone = "UTC")
    private OffsetDateTime fechaHasta;

    private Integer categoriaId;

    private String categoriaNombre = "";

    private BigDecimal categoriaBasico = BigDecimal.ZERO;

    private BigDecimal horasJornada = BigDecimal.ZERO;

    private int jornada = 0;

    private int presentismo = 0;

    private String situacion;

    private PersonaDto persona;

    private DependenciaDto dependencia;

    private CategoriaDto categoria;

}

File: src/main/java/um/haberes/report/kotlin/dto/haberes/core/CargoLiquidacionDto.kt
package um.haberes.report.kotlin.dto.haberes.core

import com.fasterxml.jackson.annotation.JsonFormat
import java.math.BigDecimal
import java.time.OffsetDateTime

data class CargoLiquidacionDto(

    var cargoLiquidacionId: Long? = null,
    var legajoId: Long? = null,
    var anho: Int? = null,
    var mes: Int? = null,
    var dependenciaId: Int? = null,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaDesde: OffsetDateTime? = null,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaHasta: OffsetDateTime? = null,
    var categoriaId: Int? = null,
    var categoriaNombre: String = "",
    var categoriaBasico: BigDecimal = BigDecimal.ZERO,
    var horasJornada: BigDecimal = BigDecimal.ZERO,
    var jornada: Int = 0,
    var presentismo: Int = 0,
    var situacion: String? = null,
    var persona: PersonaDto? = null,
    var dependencia: DependenciaDto? = null,
    var categoria: CategoriaDto? = null

)
